test_ete/test_end_to_end.py

```diff
@@ -0,0 +1,635 @@
+    rich_print(f"Testing with sklearn model: {run_sklearn}")
+    rich_print(f"Testing with tensorflow model: {run_tf}")
+    rich_print(f"Testing with pytorch model: {run_torch}")
+
+    runner = None
+    rich_print(f"CliRunner live print set to: {live_print_stdout}")
+    if live_print_stdout:
+        runner = _make_live_print_runner()
+    else:
+        runner = CliRunner()
+
+    rich_print(f"garden_ai module location: {garden_ai.__file__}")
+    rich_print("\n[bold blue]Starting ETE Test[/bold blue]")
+
+    # Cleanup any left over files generated from the test
+    _cleanup_local_files(local_files_list)
+
+    gc = None
+    if ete_grant == "cc":
+        # Create GardenClient with ClientCredentialsAuthorizer and patch all instances of GardenClients
+        rich_print("Initializing GardenClient with CC grant.")
+        if is_gha:
+            GARDEN_API_CLIENT_ID = os.getenv("GARDEN_API_CLIENT_ID")
+            GARDEN_API_CLIENT_SECRET = os.getenv("GARDEN_API_CLIENT_SECRET")
+        else:
+            if promt_for_secret:
+                GARDEN_API_CLIENT_ID = Prompt.ask(
+                    "Please enter the GARDEN_API_CLIENT_ID here "
+                ).strip()
+                GARDEN_API_CLIENT_SECRET = Prompt.ask(
+                    "Please enter the GARDEN_API_CLIENT_SECRET here "
+                ).strip()
+            else:
+                with open("./templates/git_secrets.json") as json_file:
+                    git_secrets = json.load(json_file)
+                GARDEN_API_CLIENT_ID = git_secrets["GARDEN_API_CLIENT_ID"]
+                GARDEN_API_CLIENT_SECRET = git_secrets["GARDEN_API_CLIENT_SECRET"]
+
+        gc = _auth_setup_cc(GARDEN_API_CLIENT_ID, GARDEN_API_CLIENT_SECRET)
+
+    elif ete_grant == "at":
+        # Create GardenClient normally with access token grant
+        rich_print("Initializing GardenClient with AT grant.")
+        gc = _auth_setup_at()
+    else:
+        raise Exception(
+            "Invalid grant type; must be either CC (Client credential grant) or AT (Access token grant)."
+        )
+
+    # Patch all instances of GardenClient with our new grant type one.
+    with mocker.patch(
+        "garden_ai.app.garden.GardenClient"
+    ) as mock_garden_gc, mocker.patch(
+        "garden_ai.app.model.GardenClient"
+    ) as mock_model_gc, mocker.patch(
+        "garden_ai.app.pipeline.GardenClient"
+    ) as mock_pipeline_gc:
+        mock_garden_gc.return_value = gc
+        mock_model_gc.return_value = gc
+        mock_pipeline_gc.return_value = gc
+
+        old_cwd = os.getcwd()
+        os.chdir(key_store_path)
+
+        # Garden create
+        new_garden = _test_garden_create(example_garden_data, garden_title, runner)
+
+        # Pipeline create
+        _test_pipeline_create(
+            example_pipeline_data,
+            key_store_path,
+            scaffolded_pipeline_folder_name,
+            runner,
+        )
+
+        if run_sklearn:
```

Comment:
these sections are a little verbose -- any reason not to group them together into larger `if run_{sklearn, tf, torch}:` blocks vs many small `if`s?